waddrmgr/manager_test.go

```diff
@@ -1019,38 +1019,64 @@ func testImportScript(tc *testContext) bool {
 
 // testMarkUsed ensures used addresses are flagged as such.
 func testMarkUsed(tc *testContext) bool {
-	expectedAddr1 := expectedAddr{
-		addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
-		used:        true,
+	tests := []struct {
+		name       string
+		addrType   btcutil.Address
+		in         string
+		blockstamp waddrmgr.BlockStamp
+		expected   expectedAddr
+	}{
+		{
+			name:     "managed address",
+			addrType: &btcutil.AddressPubKeyHash{},
+			in:       "1HZwkjkeaoZfTSaJxDw6aKkxp45agDiEzN",
+			expected: expectedAddr{
+				addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
+				used:        true,
+			},
+		},
+		{
+			name:     "script address",
+			addrType: &btcutil.AddressPubKeyHash{},
+			in:       "3MbyWAu9UaoBewR3cArF1nwf4aQgVwzrA5",
+			expected: expectedAddr{
+				addressHash: hexToBytes("da6e6a632d96dc5530d7b3c9f3017725d023093e"),
+				used:        true,
```

Comment:
Unnecessary for this test, it always first tests that it is unused, marks used, and then checks if it's marked used.